SingularityService/src/main/java/com/hubspot/singularity/auth/SingularityAuthorizationHelper.java

```diff
@@ -138,13 +140,14 @@ public void checkForAuthorization(SingularityRequest request, Optional<Singulari
     checkUnauthorized(user.isPresent(), "user must be present");
 
     final Set<String> userGroups = user.get().getGroups();
+    final SetView readWriteGroups = Sets.union(request.getGroup().asSet(), request.getReadWriteGroups().asSet());
```

Comment:
nitpick but `SetView<>` implements `Set<>`, so for simplicity just use a `Set<String>` here